src/Proto.Actor/Mailbox/Mailbox.cs

```diff
@@ -85,14 +85,41 @@ IMailboxQueue userMailbox
 
         public void PostUserMessage(object msg)
         {
-            _userMailbox.Push(msg);
-
-            foreach (var t in _stats)
+            // if the message is a batch message, we unpack the content as individual messages in the mailbox
+            // feature Aka: Samkuvertering in Swedish...
+            if (msg is IMessageBatch  || msg is MessageEnvelope e && e.Message is IMessageBatch)
             {
-                t.MessagePosted(msg);
+                var batch = (IMessageBatch)MessageEnvelope.UnwrapMessage(msg)!;
+                var messages = batch.GetMessages();
+
+                foreach (var message in messages)
+                {
+                    _userMailbox.Push(message);
+                    foreach (var t in _stats)
+                    {
+                        t.MessagePosted(message);
+                    }
+                }
+                
+                _userMailbox.Push(msg);
```

Comment:
the message batch message places itself also on the mailbox. as a checkpoint for the actor to know the entire batch has been consumed.
this can be leveraged for writing state back to DB for persistent actors etc. w/o hitting DB for each single message